web/src/app/components/Lobby.tsx

```diff
@@ -30,6 +32,14 @@ export default function Lobby({roomID, players, onStart, onLeave}: LobbyProps) {
         
     }
 
+    useEffect(() => {
```

Comment:
I'm also not sure if this is the best way of finding the host since this useEffect code will run everytime the players list gets updated, which can be often.